### DISCUSSION TITLE
Flax in 64bit-precision

### DISCUSSION THREAD

**Discussion by gerdm** [google/flax#1544] [Q&A]
Hi,

I'm working with a problem which requires me to work in 64bit precision. If I pass a 64bit array to Flax model, the result is cast into 32bit. Is there any way for a model to result a 64bit array?

This is a simple example of an MLP I'm considering

```python
class MLP(nn.Module):
    @nn.compact
    def __call__(self, x):
        x = nn.relu(nn.Dense(9)(x))
        x = nn.relu(nn.Dense(20)(x))
        x = nn.relu(nn.Dense(20)(x))
        x = nn.Dense(7)(x)
        return nn.softmax(x)
```

After initialising the parameters, the values are in 64bit. However, the result on call is 32bit.

Is there any way for` __call__` to return a 64bit array?

**Comment by andsteing** [comment]
Did you read this?

https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#double-64bit-precision

The code snippet is helpful, but it would be even better to add a complete Colab minimal example (because in that case anything like additional configuration mentioned in above link would be clearly defined).
